src/lem/circuit.rs

```diff
@@ -844,34 +1049,69 @@ impl Func {
                         // one constraint for the image's hash
                         num_constraints += 1;
                     }
-                    _ => todo!(),
+                    Op::Hide(..) => {
+                        // TODO
+                        globals.insert(FWrap(F::ZERO));
+                    }
+                    Op::Open(..) => {
+                        // TODO
+                        globals.insert(FWrap(F::ZERO));
+                    }
                 }
             }
             match &block.ctrl {
                 Ctrl::Return(vars) => num_constraints + 2 * vars.len(),
-                Ctrl::MatchTag(_, cases) => {
+                Ctrl::IfEq(_, _, eq_block, else_block) => {
+                    num_constraints
+                        + if nested { 6 } else { 4 }
+                        + recurse(eq_block, true, globals, store)
+                        + recurse(else_block, true, globals, store)
+                }
+                Ctrl::MatchTag(_, cases, def) => {
                     // `alloc_equal_const` adds 3 constraints for each case and
                     // the `and` is free for non-nested `MatchTag`s, since we
-                    // start `concrete_path` with a constant `true`
+                    // start `not_dummy` with a constant `true`
                     let multiplier = if nested { 4 } else { 3 };
 
                     // then we add 1 constraint from `enforce_selector_with_premise`
                     num_constraints += multiplier * cases.len() + 1;
 
                     // stacked ops are now nested
                     for block in cases.values() {
-                        num_constraints += recurse(block, true, globals);
+                        num_constraints += recurse(block, true, globals, store);
                     }
+                    match def {
+                        Some(def) => {
+                            // constraints for the boolean and the default case
+                            num_constraints += multiplier - 2;
```

Comment:
How come?

Reply:
Reason: no `alloc_eq` which costs 3, but one allocated bit which costs 1, i.e. -2 compared to the other branches

Reply:
Sorry, I meant that these could be added to the source code as commentaries, like what we have for the other cases